Add the RELEASES.md for explaining the way to release (#3807)
pipe-cd/pipecd: RELEASES.md

# Release Process
This document explains the process to release new version and to address patch release.

## Major release
This refers to the release of new features.

### Confirm the changelog and Create Release Note
- Run the release script

  This example assumes that `vX.Y.Z` will be released:
  ```shell
  make gen/release version=vX.Y.Z
  ````

  `RELEASE` file will be updated and `docs/content/en/blog/releases/vX.Y.Z.md` file will be created.

- Push the above changes and Create a pull request to confirm the changelog.
  You can confirm the changelog through the reviewing comment in pull request by GitHub Actions.
  For more information, Please see [actions-gh-release](https://github.com/pipe-cd/actions-gh-release).

- Update the content in `docs/content/en/blog/releases/vX.Y.Z.md` file based on changelog results.
  Please refer to [this example](https://github.com/pipe-cd/pipecd/pull/3792/commits/2b59f7f2a492405bf6df905b3823b417e4a10c3e).

  It is recommended to commit the above changes once here.

### Generate document for new version
- Run the release document script

  This example assumes that `vX.Y.Z` will be released:
  ```shell

>   make release-docs version=vX.Y.Z

Reviewer: ```suggestion
  make gen/release-docs version=vX.Y.Z
```

  ````

> - Push the above changes and get a review and merge.

Reviewer: ```suggestion
- Make a pull request to `master` branch with the above changes and get reviews and merge.
```

### Cut a new release
- Before cutting a new release, wait for all jobs in GitHub Actions to pass on master branch.

- Create a tagged release. The release should start with "v" and be followed by the version number.

- Create a branch from the tagged release, e.g. "release-X.Y". This will be used for the patch release.

## Patch release

> This refers to the release of critical bug fixes.

Reviewer: ```suggestion
This refers to the release of critical bug fixes. \
```
nits

Author: Sorry, I'm not sure about this meaning. Could you explain that difference?

Author: Ah, I see. Thanks!

A bugfix for a functional issue (not a data loss or security issue) that only affects an alpha feature does not qualify as a critical bug fix.

### Prerequisites
- The release branch exists, e.g. "release-X.Y".
- (optional) If you create a pull request with the cherry-pick script, `gh` is needed to be installed and ran `gh auth login`. Please refer to [cli/cli](https://github.com/cli/cli).

### Fix bugs
- Create a pull request to fix a bug on the `master` branch.

> - Get a review and merge.

Reviewer: ```suggestion
- Get reviews and merge.
```

# Release Process

> *This release process have not been fixed yet, hence this is subject to frequent change.*

Reviewer 2: Should we remove this sentence?

Author: Right, Thanks!

This document explains the process to release new version and to address stable release.

## Major release
This refers to the release of new features.

### Confirm the changelog and Create Release Note
- Run the release script

  This example assumes that `vX.Y.Z` will be released:
  ```shell
  make gen/release version=vX.Y.Z
  ````

  `RELEASE` file will be updated and `docs/content/en/blog/releases/vX.Y.Z.md` file will be created.

- Push the above changes and Create a pull request to confirm the changelog.
  You can confirm the changelog through the reviewing comment in pull request by GitHub Actions.
  For more information, Please see [actions-gh-release](https://github.com/pipe-cd/actions-gh-release).

- Update the content in `docs/content/en/blog/releases/vX.Y.Z.md` file based on changelog results.
  Please refer to [this example](https://github.com/pipe-cd/pipecd/pull/3792/commits/2b59f7f2a492405bf6df905b3823b417e4a10c3e).

  It is recommended to commit the above changes once here.

> ### Create Document

Reviewer 2: ```suggestion
### Generate Document for new version
```

- Run the release document script

  This example assumes that `vX.Y.Z` will be released:
  ```shell
  make release-docs version=vX.Y.Z
  ````

- Push the above changes and get a review and merge.

### Cut a new release
- Before cutting a new release, wait for all jobs in GitHub Actions to pass on master branch.

- Create a tagged release. The release should start with "v" and be followed by the version number.

- Create a branch from the tagged release, e.g. "release-X.Y". This will be used for the stable release.

> ## Stable release

Reviewer 2: How about `Patch release`?
